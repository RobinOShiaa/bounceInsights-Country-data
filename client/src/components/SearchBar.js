// Input field within the sidebar for searching for country by name
const SearchBar = (props) => {
  const { onChange } = props;
  return (
    <div className="mb-2 -mx-2">
      <label
        htmlFor="email"
        className="block pt-8 text-xs font-semibold leading-6 text-indigo-200"
      >
        Countries
      </label>
      <div className="mt-2">
        <input
          data-testid="test-input"
          type="text"
          name="country"
          id="countrySearch"
          className="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
          placeholder="Ireland"
          onChange={(e) => onChange(e)}
        />
      </div>
    </div>
  );
};

export default SearchBar;
