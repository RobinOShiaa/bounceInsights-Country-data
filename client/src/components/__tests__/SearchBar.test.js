import SearchBar from "../SearchBar";
import { fireEvent, render } from "@testing-library/react";

const mock = jest.fn(() => {});

test("Matches SearchBar component snapshot", () => {
  const tree = render(<SearchBar onChange={mock} />);
  expect(tree).toMatchSnapshot();
});

test("Ensures function is called for onChange", () => {
  const utils = render(<SearchBar onChange={mock} />);
  const input = utils.getByTestId("test-input");
  fireEvent.change(input, { target: { value: "23" } });
  expect(input.value).toBe("23");
});
