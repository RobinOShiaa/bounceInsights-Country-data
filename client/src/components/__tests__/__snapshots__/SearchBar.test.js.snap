// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches SearchBar component snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mb-2 -mx-2"
      >
        <label
          class="block pt-8 text-xs font-semibold leading-6 text-indigo-200"
          for="email"
        >
          Countries
        </label>
        <div
          class="mt-2"
        >
          <input
            class="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
            data-testid="test-input"
            id="countrySearch"
            name="country"
            placeholder="Ireland"
            type="text"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mb-2 -mx-2"
    >
      <label
        class="block pt-8 text-xs font-semibold leading-6 text-indigo-200"
        for="email"
      >
        Countries
      </label>
      <div
        class="mt-2"
      >
        <input
          class="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
          data-testid="test-input"
          id="countrySearch"
          name="country"
          placeholder="Ireland"
          type="text"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
